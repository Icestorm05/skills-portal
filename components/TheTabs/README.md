# The Tabs

## Functionality

The tabs component displays a list of routes available in the application. These routes include: `Your Profile`, `Your Skills`, `Your Available Skills` and `Your Team` (optional). This component should only be used once within the lifecycle of this application.

## Usage

### Props

`<the-tabs>` supports the following custom component attributes:

| attribute | type | description
| --- | --- | ---
| `color` | String *optional* | The color of each tab. Defaults to `secondary`. Accepts any of: `primary`, `accent`, `secondary`, `info`, `warning`, `error` or `success`.
| `hasTeam` | Boolean *optional* | Whether or not to display the team tab or not. Defaults to `false`.
| `sliderColor` | String *optional* | The slider color of the selected tab. Defaults to `warning`. Accepts any of: `primary`, `accent`, `secondary`, `info`, `warning`, `error` or `success`.

### Example

```html
<the-tabs/>
```

```ts
import Component from 'nuxt-class-component';
import Vue from 'vue';

import TheTabs from '~/components/TheTabs/TheTabs.vue';

@Component({
    components: {
        TheTabs,
    },
})
export default class GenericComponent extends Vue {}
```