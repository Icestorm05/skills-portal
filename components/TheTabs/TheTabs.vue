<template lang="pug">
    v-tabs(:color="color"
           grow
           icons-and-text
           show-arrows)#TheTabs
        v-tabs-slider(:color="sliderColor")
        v-tab(ripple
              nuxt
              exact
              :to="{ name: 'id-profile' }").white--text.Tab#ProfileTab Your Profile
              v-icon person
        v-tab(ripple
              nuxt
              exact
              :to="{ name: 'id-skills' }").white--text.Tab#SkillsTab Your Skills
              v-icon list
        v-tab(ripple
              nuxt
              exact
              :to="{ name: 'id-skills-available' }").white--text.Tab#AvailableSkillsTab Skills Catalogue
              v-icon playlist_add
        v-tab(ripple
              nuxt
              exact
              :to="{ name: 'id-team' }"
              v-if="hasTeam").white--text.Tab#TeamTab Your Team
              v-icon people
</template>

<script lang="ts">
import Component from "nuxt-class-component";
import Vue from "vue";
import { Prop } from "vue-property-decorator";

import VIcon from "vuetify/es5/components/VIcon";
import VTabs, { VTab, VTabsSlider } from "vuetify/es5/components/VTabs";

@Component({
  components: {
    VIcon,
    VTab,
    VTabs,
    VTabsSlider,
  }
})
export default class TheUserTabs extends Vue {
  @Prop({ type: String, default: "secondary" })
  private color: string;
  @Prop({ type: Boolean, default: false })
  private hasTeam: boolean;
  @Prop({ type: String, default: "warning" })
  private sliderColor: string;
}
</script>

<style lang="stylus" scoped>
.Tab:hover /deep/ .tabs__item {
    opacity: 1;
}
</style>
