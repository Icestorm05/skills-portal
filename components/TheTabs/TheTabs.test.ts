import TheTabs from "./TheTabs.vue";
import Vuetify from "vuetify/es5/components/Vuetify";
import VTabs from "vuetify/es5/components/VTabs";
import { createLocalVue, mount } from "@vue/test-utils";

declare const expect: jest.Expect;

const localVue = createLocalVue();
localVue.use(Vuetify, {
  components: {
    VTabs
  }
});

const stubs = {
  "nuxt-link": "<a></a>"
};

describe("The Tabs", () => {
  test("is a Vue instance", () => {
    const wrapper = mount(TheTabs, { localVue, stubs });
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  test("snapshot has not changed", () => {
    const wrapper = mount(TheTabs, { localVue, stubs });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("color prop changes color of tabs", () => {
    const wrapper = mount(TheTabs, {
      localVue,
      stubs,
      propsData: { color: "error" }
    });
    expect(wrapper.find("#TheTabs .tabs__bar").classes()).toContain("error");
  });

  test("slider color prop changes color of tab slider", () => {
    const wrapper = mount(TheTabs, {
      localVue,
      stubs,
      propsData: { sliderColor: "error" }
    });
    expect(wrapper.find("#TheTabs .tabs__slider").classes()).toContain("error");
  });

  test("team tab is hidden by default", () => {
    const wrapper = mount(TheTabs, { localVue, stubs });
    expect(wrapper.find("#TheTabs #TeamTab").exists()).toBeFalsy();
  });

  test("team tab is visible with hasTeam prop", () => {
    const wrapper = mount(TheTabs, {
      localVue,
      stubs,
      propsData: { hasTeam: true }
    });
    expect(wrapper.find("#TheTabs #TeamTab").exists()).toBeTruthy();
  });
});
