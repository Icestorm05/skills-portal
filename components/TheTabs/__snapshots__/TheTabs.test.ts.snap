// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Tabs snapshot has not changed 1`] = `
<div class="tabs" id="TheTabs">
    <div class="tabs__bar secondary">
        <!---->
        <div class="tabs__wrapper">
            <div class="tabs__container tabs__container--grow tabs__container--icons-and-text">
                <div class="tabs__slider-wrapper">
                    <div class="tabs__slider warning"></div>
                </div>
                <div class="tabs__div white--text Tab" id="ProfileTab">
                    <nuxt-link class="tabs__item">Your Profile
                        <i aria-hidden="true" class="icon material-icons">person</i>
                    </nuxt-link>
                </div>
                <div class="tabs__div white--text Tab" id="SkillsTab">
                    <nuxt-link class="tabs__item">Your Skills
                        <i aria-hidden="true" class="icon material-icons">list</i>
                    </nuxt-link>
                </div>
                <div class="tabs__div white--text Tab" id="AvailableSkillsTab">
                    <nuxt-link class="tabs__item">Skills Catalogue
                        <i aria-hidden="true" class="icon material-icons">playlist_add</i>
                    </nuxt-link>
                </div>
                <!---->
            </div>
        </div>
        <!---->
    </div>
</div>
`;
